src/btree/row_srch.c

```diff
@@ -651,11 +654,15 @@ __wt_row_search(WT_CURSOR_BTREE *cbt, WT_ITEM *srch_key, bool insert, WT_REF *le
          * Reset the skipped prefix counts; we'd normally expect the parent's skipped prefix values
          * to be larger than the child's values and so we'd only increase them as we walk down the
          * tree (in other words, if we can skip N bytes on the parent, we can skip at least N bytes
-         * on the child). However, leaf pages at the end of the tree can be extended, causing the
-         * parent's search to be wrong for the child. We only need to reset the high count, the page
-         * can only be extended so the low count is still correct.
+         * on the child). However, we can insert smaller (larger) keys to leaf pages at the start
+         * (end) of the tree, causing the parent's search to be wrong for the child. In this case,
+         * we need to reset the boundary.
+         *
+         * It is not easy to detect which case we are exactly in with the available information.
+         * Simply reset both the low and high boundary if we detect a change of tree structure.
          */
-        skiphigh = 0;
```

Comment:
Is it ok to move `skiphigh = 0` inside the if branch here? Looking at [our description of WT-6257](https://jira.mongodb.org/browse/WT-10961?focusedCommentId=5416622&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-5416622) it seems like we could hit WT-6257 without the pindex changing as we only need to delete the rightmost page and insert one more key to hit the issue.

The same question applies to the internal page logic on line 506 if this is accurate.